File: JavaScript/src/report.js
function report(string) {
  let output = [];
  let grades = string.split(",");
  
  let greens = grades.filter(grade => grade.toLowerCase().trim() === "green").length;
  if (greens > 0) { output.push(`Green: ${greens}`) };
  
  let ambers = grades.filter(grade => grade.toLowerCase().trim() === "amber").length;
  if (ambers > 0) { output.push(`Amber: ${ambers}`) };
  
  let reds = grades.filter(grade => grade.toLowerCase().trim() === "red").length;
  if (reds > 0) { output.push(`Red: ${reds}`) };

  return output.join("\n");
}

module.exports = report;

File: Ruby/lib/report.rb
def report(input)
  arr = input.split(',').map(&:strip)
  output = []
  
  output.push("Green: #{arr.count('Green')}") if arr.count('Green') > 0
  
  output.push("Amber: #{arr.count('Amber')}") if arr.count('Amber') > 0
  
  output.push("Red: #{arr.count('Red')}") if arr.count('Red') > 0
  
  other = arr.length - (arr.count('Green') + arr.count('Amber') + arr.count('Red'))
  output.push("Other: #{other}") if other > 0

  output.join("\n")
end

  
